src/lib/functions/registry.ts

```diff
@@ -434,7 +434,7 @@ export class FunctionsRegistry {
 
         func.mainFile = v2EntryPointPath
       } catch {
-        func.mainFile = join(unzippedDirectory, `${func.name}.js`)
+        func.mainFile = join(unzippedDirectory, basename(manifestEntry.mainFile))
```

Comment:
the problem here was that esm files end in `.mjs`, not `.js`. By using the basename from the manifest, we'll have the right extension